### ISSUE TITLE
AllOf with enum  generates code 

### ISSUE THREAD

**Issue by mamiksik** [anymaniax/orval#339]
## What are the steps to reproduce this issue?
1. Schema containing enum and allOf
![image](https://user-images.githubusercontent.com/6180007/156580077-23e8407b-6e44-4788-a544-99efa8868693.png)
![image](https://user-images.githubusercontent.com/6180007/156580250-6dfd58fd-752b-4138-9ffa-f8a5e5d6e2aa.png)

2. Generate types

## What happens?
Generated code is invalid

![image](https://user-images.githubusercontent.com/6180007/156580830-704f0df2-6bea-4cdf-86a1-87d1250d04a3.png)

## What were you expecting to happen?
`export const SourceFileStatus = FileStatusEnum;` should be added after the closing bracket of an interface, not in front of it. 

Like this:
![image](https://user-images.githubusercontent.com/6180007/156583193-9f39f68b-1c83-4c1d-a715-11d4d6564281.png)

## Any other comments?
For enum definition without `allOf` the generated code is correct

Such as:
![image](https://user-images.githubusercontent.com/6180007/156582007-87c7cc0f-8186-4420-ac87-c0d17f5f451e.png)

## What versions are you using?
- Operating System: *Linux (Ubuntu 21.10)*
- Package Version:*6.6.4*
- OpenApi v3

Hope everything is clear, if not just tag me. I will happly add any additional information:)

**Comment by anymaniax** [comment]
Should be fixed in the next version

**Comment by mamiksik** [comment]
Thanks! btw: Thumbs up for this amazing lib. I love using it!

**Comment by tinybrain** [comment]
If this is wrt #330, then there I'm still having an issue (working from master).

For me it's with enum properties in inherited classes (c# NSwag).

The `combineSchemas()` is getting called when iterating through the properties, and it's adding the `newEnum` export to the end of the property, not the end of the model declaration, so that combined type needs to be deferred somehow.

Schema (bottom) produces:

testModelAllOf.ts

```ts
import { TestType } from './testType';

export type TestModelAllOf = {
  id: number;
  name: string;
  type: TestType; // <- should be TestModelAllOfType

// eslint-disable-next-line @typescript-eslint/no-redeclare
export const TestModelAllOfType = TestType;
};
```

testType.ts

```ts
export type TestType = 'RUBBER' | 'BABY' | 'BUNKER';


// eslint-disable-next-line @typescript-eslint/no-redeclare
export const TestType = {
  RUBBER: 'RUBBER' as TestType,
  BABY: 'BABY' as TestType,
  BUNKER: 'BUNKER' as TestType,
};
```

As a workaround I'm just declaring the property's combined type without an export, since even though all my enums are `allOf` and `isAllEnums`, they're `allOf` 1 enum so I don't need the export (see 79466ca17b7b397f2f1398129b157b370c0dce33)

Schema:

```json
{
  "swagger": "2.0",
  "info": {
    "title": "Enum Test",
    "version": "1.0.0"
  },
  "schemes": ["https"],
  "consumes": ["application/json"],
  "produces": ["application/json"],
  "paths": {
    "/api/v1/system/test/{id}": {
      "get": {
        "tags": ["Test"],
        "operationId": "Test_GetTest",
        "parameters": [
          {
            "type": "integer",
            "name": "id",
            "in": "path",
            "required": true,
            "format": "int32",
            "x-nullable": false
          }
        ],
        "responses": {
          "200": {
            "x-nullable": true,
            "description": "",
            "schema": {
              "$ref": "#/definitions/TestModel"
            }
          }
        }
      }
    }
  },
  "definitions": {
    "TestModel": {
      "allOf": [
        {
          "$ref": "#/definitions/TestModelBase"
        },
        {
          "type": "object",
          "required": ["id", "name", "type"],
          "properties": {
            "id": {
              "title": "ID",
              "type": "integer",
              "format": "int32"
            },
            "name": {
              "title": "Name",
              "type": "string",
              "maxLength": 255,
              "minLength": 0
            },
            "type": {
              "title": "Type",
              "allOf": [
                {
                  "$ref": "#/definitions/TestType"
                }
              ]
            }
          }
        }
      ]
    },
    "TestType": {
      "type": "string",
      "description": "",
      "x-enumNames": ["Rubber", "Baby", "Bunker"],
      "enum": ["RUBBER", "BABY", "BUNKER"]
    },
    "TestModelBase": {
      "type": "object",
      "x-abstract": true
    }
  }
}
```

**Comment by anymaniax** [comment]
Fixed with `6.7.0`

**Comment by o-alexandrov** [comment]
@mamiksik consider to close, if it's fixed
